src/sched.rs

```diff
@@ -140,6 +140,13 @@ mod sched_linux_like {
 
         Errno::result(res).map(drop)
     }
+
+    /// Determines the CPU on which the calling thread is running.
+    pub fn sched_getcpu() -> Result<usize> {
```

Comment:
It looks like this is supported on other platforms, including [FreeBSD](https://github.com/freebsd/freebsd-src/blob/8b43388c745ebb42414496f2925f47acac662c2f/include/sched.h#L45) and [DragonFly](https://github.com/DragonFlyBSD/DragonFlyBSD/blob/39b691836ec8de6e84bc13067f04db947dd032f6/sys/sys/sched.h#L84).

Can you move this to the `sched_affinity` section below (since it seems related and has these platforms already?